docs/content/guides/developer/app-examples/price-oracle.mdx

```diff
@@ -0,0 +1,204 @@
+---
+title: Oracles
+---
+
+This reference guide presents various Oracles that you can utilize while building on Sui. Oracles supply offchain data to the blockchain, enabling smart contracts to access a diverse range of information.
+
+## Pyth Network
+
+The [Pyth Network](https://pyth.network/) is one of the largest first-party Oracle network, delivering real-time data across [a vast number of chains](https://docs.pyth.network/price-feeds/contract-addresses).
+
+The network comprises some of the world’s [largest exchanges, market makers, and financial services providers](https://pyth.network/publishers). These publish proprietary data on-chain for aggregation and distribution to smart contract applications. 
+
+## How to Use Real-Time Data in Sui Contracts using Pyth Oracle
+
+This guide explains how to use real-time Pyth data in Sui applications.
+
+## Install Pyth SDK
+
+Use the following dependency in your `Move.toml` file to use the latest Pyth Sui package and its dependencies:
+
+```bash title='Sui Mainnet'
+[dependencies.Pyth]
+git = "https://github.com/pyth-network/pyth-crosschain.git"
+subdir = "target_chains/sui/contracts"
+rev = "sui-contract-mainnet"
+
+[dependencies.Wormhole]
+git = "https://github.com/wormhole-foundation/wormhole.git"
+subdir = "sui/wormhole"
+rev = "sui-upgrade-mainnet"
+
+# Pyth is locked into this specific `rev` because the package depends on Wormhole and is pinned to this version.
+
+[dependencies.Sui]
+git = "https://github.com/MystenLabs/sui.git"
+subdir = "crates/sui-framework/packages/sui-framework"
+rev = "041c5f2bae2fe52079e44b70514333532d69f4e6"
+
+```
+
+```bash title='Sui Testnet'
+[dependencies.Pyth]
+git = "https://github.com/pyth-network/pyth-crosschain.git"
+subdir = "target_chains/sui/contracts"
+rev = "sui-contract-testnet"
+
+[dependencies.Wormhole]
+git = "https://github.com/wormhole-foundation/wormhole.git"
+subdir = "sui/wormhole"
+rev = "sui-upgrade-testnet"
+
+# Pyth is locked into this specific `rev` because the package depends on Wormhole and is pinned to this version.
+[dependencies.Sui]
+git = "https://github.com/MystenLabs/sui.git"
+subdir = "crates/sui-framework/packages/sui-framework"
+rev = "041c5f2bae2fe52079e44b70514333532d69f4e6"
+```
+
+
+Pyth also provides a javascript SDK to construct transaction blocks that update price feeds:
+
+    ```bash title='npm'
+    npm install --save @pythnetwork/pyth-sui-js
+    ```
+ 
+     ```bash title='yarn'
+    yarn add @pythnetwork/pyth-sui-js
+    ```
+
+
+## Write Contract Code
+
+The code snippet below provides a general template for what your contract code should look like:
+
+```rust
```

Comment:
Can you please make sure the code follows Move 2024 styles? Utilising module label, calling methods on structs, and following convention in error naming?

```move
module pyth_example::main;

use sui::clock::Clock;

// ...
const EInvalidId: u64 = 1;
```

Reply:
Also, using `move` as the language tag.